File: solutions/opus-1/opus/c/primes.c
#include <stdio.h>
#include <string.h>

static char composite[100001];

int main(void) {
    int n;
    if (scanf("%d", &n) != 1) n = 0;
    if (n < 0) n = 0;
    if (n > 100000) n = 100000;
    int first = 1;
    for (int i = 2; i <= n; i++) {
        if (composite[i]) continue;
        if (!first) putchar(' ');
        printf("%d", i);
        first = 0;
        for (long long j = (long long)i * i; j <= n; j += i) composite[j] = 1;
    }
    putchar('\n');
    return 0;
}

File: solutions/opus-1/opus/go/primes.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	var n int
	fmt.Fscan(bufio.NewReader(os.Stdin), &n)
	w := bufio.NewWriter(os.Stdout)
	defer w.Flush()
	composite := make([]bool, n+1)
	first := true
	for i := 2; i <= n; i++ {
		if composite[i] {
			continue
		}
		for j := i * i; j <= n; j += i {
			composite[j] = true
		}
		if !first {
			w.WriteByte(' ')
		}
		w.WriteString(strconv.Itoa(i))
		first = false
	}
	w.WriteByte('\n')
}
